### PR TITLE
[Bug fix] The bug following be fixed:

### PR DESCRIPTION
**Pull request by HappenLee**
## Proposed changes

     1. Datetime/Date cast string add invalid '/0'
     2. Decimal support mod, muliply, plus, minus
     3. function concat_ws stack over flow
     4. bigint compute failed because of return int128
     5. dead lock cause by stream reciver manager do not close
     6. conjunts push down cause error result
     7. String cast date bug
     8. filter block do filter the right column
     9. datetime expr computer expr bug
     10. Exchange Node Support limit
     11. DataTime support change to Int
     12. fix bug of mem_reuse of union node
     13. change some bool function return type
     14. if function rewrite
     15. disable runtime filter in vec exec engine
     16. only support inner join in hash join node
     17. Function diff/negative bug
     18. mysql result writer overflow the buffer
     19. Fix some math function get wrong return type

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exec/olap_scan_node.cpp [hunk 1]
```diff
@@ -65,7 +65,7 @@ OlapScanNode::~OlapScanNode() {}
 
 Status OlapScanNode::init(const TPlanNode& tnode, RuntimeState* state) {
     RETURN_IF_ERROR(ExecNode::init(tnode, state));
-    _direct_conjunct_size = state->enable_vectorized_exec() ? 1 : _conjunct_ctxs.size();
+    _direct_conjunct_size =  _conjunct_ctxs.size();
```
Comment (wangbo) [inline be/src/exec/olap_scan_node.cpp:68]: 这个地方向量化版和非向量化版的区别在于，fe生成的ExprContext个数不同，之前向量化版是生成一个，你要改这里的话，是不是fe生成谓词树的语法也得改下

Comment (HappenLee) [reply be/src/exec/olap_scan_node.cpp:68]: 嗯，这个其实只涉及到谓词下推到存储引擎的逻辑。因为当前谓词下推的时候都是基于原有的_conjunt_ctxs处理，这里使用非向量化版本的vconjunts计数可能会导致一些问题

File: be/src/vec/data_types/data_type_date.cpp [hunk 2]
```diff
@@ -51,7 +51,8 @@ void DataTypeDate::to_string(const IColumn & column, size_t row_num, BufferWrita
 
     char buf[64];
     char* pos = value.to_string(buf);
-    ostr.write(buf, pos - buf);
+    // DateTime to_string the end is /0
+    ostr.write(buf, pos - buf - 1);
```
Comment (wangbo) [inline be/src/vec/data_types/data_type_date.cpp:55]: 好奇问下，为啥旧版的datetime tostring的时候，末尾为啥要加0，是有啥考虑吗

File: be/src/vec/functions/function_math_unary.h [hunk 3]
```diff
@@ -140,14 +140,15 @@ class FunctionMathUnary : public IFunction {
...
     static constexpr auto name = Name::name;
     static constexpr auto rows_per_iteration = 1;
-    static constexpr bool always_returns_float64 = true;
+    static constexpr bool always_returns_float64 = std::is_same_v<Type, DataTypeFloat64>;
 
-    template <typename T>
-    static void execute(const T* src, Float64* dst) {
+    template <typename T, typename U>
+    static void execute(const T* src, U* dst) {
         dst[0] = static_cast<Float64>(Function(static_cast<Float64>(src[0])));
```
Comment (wangbo) [inline be/src/vec/functions/function_math_unary.h:152]: 这里cast的返回值类型是不是得和ReturnType保持一致，而不是Float64

File: be/src/vec/functions/function_math_unary.h [hunk 4]
```diff
@@ -124,7 +124,7 @@ class FunctionMathUnary : public IFunction {
             using Types = std::decay_t<decltype(types)>;
             using Type = typename Types::RightType;
             using ReturnType = std::conditional_t<
-                    Impl::always_returns_float64 || !std::is_floating_point_v<Type>, Float64, Type>;
+                    Impl::always_returns_float64, Float64, Int64>;
```
Comment (wangbo) [inline be/src/vec/functions/function_math_unary.h:127]: 这里写死Int64应该有问题，types入参是会有int128的，还是得用Type吧

File: be/src/vec/functions/function_string.cpp [hunk 5]
```diff
@@ -482,7 +482,7 @@ void register_function_string(SimpleFunctionFactory& factory) {
     factory.register_function<FunctionStringEndsWith>();
     factory.register_function<FunctionStringInstr>();
     factory.register_function<FunctionStringFindInSet>();
-    factory.register_function<FunctionStringLocate>();
+//    factory.register_function<FunctionStringLocate>();
```
Comment (wangbo) [inline be/src/vec/functions/function_string.cpp:485]: 这个函数为啥注释了

File: be/src/vec/functions/function_string.h [hunk 6]
```diff
@@ -558,9 +558,9 @@ class FunctionStringPad : public IFunction {
     size_t get_number_of_arguments() const override { return 3; }
 
     DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
-        return make_nullable(std::make_shared<DataTypeString>());
+        return std::make_shared<DataTypeString>();
```
Comment (wangbo) [inline be/src/vec/functions/function_string.h:561]: FunctionStringPad的execute_impl返回类型是一个强转成nullable了，你这里把nullable移除了不就对不上了吗

File: be/src/vec/functions/if.cpp [hunk 7]
```diff
@@ -396,10 +396,25 @@ class FunctionIf : public IFunction {
         bool cond_is_null = arg_cond.column->only_null();
 
         if (cond_is_null) {
-            block.replace_by_position(result, arg_else.column);
+            block.replace_by_position(result, arg_else.column->clone_resized(arg_cond.column->size()));
```
Comment (wangbo) [inline be/src/vec/functions/if.cpp:399]: 这个地方不clone会有啥问题吗

File: be/src/vec/functions/if.cpp [hunk 8]
```diff
@@ -396,10 +396,25 @@ class FunctionIf : public IFunction {
...
             return true;
         }
 
+        if (auto * nullable = check_and_get_column<ColumnNullable>(*arg_cond.column))
+        {
```
Comment (wangbo) [inline be/src/vec/functions/if.cpp:404]: 这里感觉可以加一个DCheck，预期的嵌套字段的实际类型应该是uint8
另外这里的大括号需要放到上一行

File: be/src/vec/data_types/data_type_date.cpp (hunk 2 above)
Comment (HappenLee) [reply be/src/vec/data_types/data_type_date.cpp:55]: 因为那个返回的结果是char*，这部分代码好像是直接从mysql里面copy过来的。我的理解是保持了c语言字符串的状态，所以填了一个\0

File: be/src/vec/functions/function_math_unary.h (hunk 3 above)
Comment (HappenLee) [reply be/src/vec/functions/function_math_unary.h:152]: 这个有个问题，ceil，floor的返回值不是float64，导致出现core的情况。所以这里计算的时候使用float64，写入的时候需要转回原始的类型，否则结果不对，会出core

File: be/src/vec/functions/function_math_unary.h (hunk 4 above)
Comment (HappenLee) [reply be/src/vec/functions/function_math_unary.h:127]: 目前只会有两个数学函数，ceil floor用到这个函数。返回值在FE是确定的int64，所以没有问题，如果是int128会cast到float64作为入参传入。

File: be/src/vec/functions/function_string.cpp (hunk 5 above)
Comment (HappenLee) [reply be/src/vec/functions/function_string.cpp:485]: 这个函数实现有些问题，短期内不好改，所以先禁用了

File: be/src/vec/functions/function_string.h (hunk 6 above)
Comment (HappenLee) [reply be/src/vec/functions/function_string.h:561]: 这个要和python脚本里定义的一直，这个函数并不是always_nullable的，所以不需要强转

File: be/src/vec/functions/if.cpp (hunk 7 above)
Comment (HappenLee) [reply be/src/vec/functions/if.cpp:399]: 在mem_reuse时会导致多指针引用，导致内存复用出现一些问题

File: be/src/vec/functions/if.cpp (hunk 8 above)
Comment (HappenLee) [reply be/src/vec/functions/if.cpp:404]: 好的，这个是clickhouse那边直接copy过来的，不知道原来为什么没有这个代码。会导致出错

**Review by yangzhg — APPROVED** [review]
LGTM

**Pull request merged**